widget/embedded/src/utils/providers.ts

```diff
@@ -107,15 +102,22 @@ function pickVersionWithFallbackToLegacy(
 ): BothProvidersInterface[] {
   const { experimentalWallet = 'enabled' } = options || {};
 
-  return providers.map((provider) => {
-    const version = experimentalWallet == 'disabled' ? '0.0.0' : '1.0.0';
-    try {
-      return pickVersion(provider, version)[1];
-    } catch {
-      // Fallback to legacy version, if target version doesn't exists.
-      return pickVersion(provider, '0.0.0')[1];
-    }
-  });
+  return providers.map((provider) =>
+    pickSingleProviderVersionWithFallbackToLegacy(provider, experimentalWallet)
+  );
+}
+
+function pickSingleProviderVersionWithFallbackToLegacy(
```

Comment:
remove the `pickVersionWithFallbackToLegacy` and change the second param to accept `options` as input.

Reply:
Done